src/libvideo/Helpers/Html.cs

```diff
@@ -14,5 +14,19 @@ internal static class Html
             WebUtility.HtmlDecode(
                 Text.StringBetween(
                     '<' + name + '>', "</" + name + '>', source));
+        
+        public static IEnumerable<string> GetUrisFromManifest(string source)
+        {
+            string opening = "<BaseURL>";
+            string closing = "</BaseURL";
+            int start = source.IndexOf(opening);
+            if (start != -1)
+            {
+                string temp = source.Substring(start);
+                var Uris = temp.Split(new string[] { opening }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(v => v.Substring(0, v.IndexOf(closing)));
+                return Uris;
+            }
+            throw new NotSupportedException();
```

Comment:
Question: Have you actually tried building/testing on your machine? It appears this is missing a closing brace. (To build you can open the `libvideo.sln` solution in VS + run NuGet restore and build.)